# AFS-Lab-scripts
The collection of scripts I've made for the Anne Ferguson-Smith Lab at the University of Cambridge, Genetics Department. Made under the direction of Dr. Kevin Costello.

----------------------------------------

Below is a list of all scripts in alphabetical order and a brief outline of what they do, what directory is works in/paths to, example usages, and outputs
Deprecated scripts are labeled as such and have ! at the beginning of thier name. They can be deleted without issue if need be

It is HIGHLY RECOMMENDED to have all necessary conda enviornments downloaded prior to usage of any script
Go to conda-envs for the .yml files and use "conda create -f name-of-enviornment.yml" in command line to download

1d1-aligner-for-all-subspecies-tree.sh

    - Synopsis: generates the multi-strain .fastas and .beds found in ../hybrids. It additionally returns an ordered list of Id1's and sequence alignments using IDs in a tree_list file (generated by run-mafft-cluster and its variants) ordered 
    - Directory info: sets working directory to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains/hybrids
    - Example usage: bash path-to-scripts/1d1-aligner-for-all-subspecies-tree.sh
    - Outputs: convert-name-all_subspecies.ordered.bed, msa-all_subspecies.ordered.txt, LTR1s-1d1-all_subspecies-ordered_matrix.txt

1d1-pop_alpha-barchart.R

    - Synopsis: generates a bar chart of the number of LTR1s in pop alpha, are 1d1's, or both
    - Directory info: uses cwd (current working directory)
    - Example usage: conda activate -n R Rscript path-to-scripts/1d1-pop_alpha-barchart.R
    - Outputs: pop_alpha-Id1-barchart.png

1d1-pop_alpha-barchart.sh 

    - Synopsis: uses windowed_tiplabs.txt and LTR1s-1d1-all_subspecies-ordered_matrix.txt to generate in-file for 1d1-pop_alpha-barchart.R and runs it 
    - Directory info: sets cwd to ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/hybrids
    - Example usage: bash path-to-scripts/1d1-pop_alpha-barchart.sh
    - Outputs: pop_alpha-Id1-barchart.png 

1delta1-condenser.R 

    - Synopsis: takes in a bed6 named "in-file.txt" and condenses TEs into one element, i.e. LTR + EZ-int + EZ-int + LTR -> LTR_TE
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/1delta1-condenser.R  
    - Outputs: outfile.txt

1delta1_reader.R 

    - Synopsis: takes in a fasta.out, pre-filtered for IAPLTR1_Mm and IAPEZ-int, named "in-file.txt" and returns an uncondensed fasta.out of 1d1 elements
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/1delta1_reader.R  
    - Outputs: temp.txt

! bed_to_LTR1_fasta.sh [DEPRECATED: use better-bed_to_tree.sh]

    - Synopsis: converts a fasta.out into a .fa of LTR1 elements
                calls name_adder_for_mafft.R internally
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains/$name
    - Example usage: bash path-to-scripts/bed_to_LTR1_fasta.sh name 
    - Outputs: IAPLTR1-TE-$name.bed, all-IAPLTRs-$name.bed, LTR1s-$name.bed, LTR1s-$name.fa

! bed_to_tree.sh [DEPRECATED: use better-bed_to_tree.sh]

    - Synopsis: combination of bed_to_LTR1_fasta.sh and fasta_to_tree.sh, looping through mouse-strains-list.txt
                calls name_adder_for_mafft.R internally
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains/$name
    - Example usage: bash path-to-scripts/bed_to_tree.sh
    - Outputs: see bed_to_LTR1_fasta.sh and fasta_to_tree.sh 

better-bed_to_tree.sh

    - Synopsis: looping through mouse-strains-list.txt, runs better-IAPLTR-1d1-puller.sh then pulls whole LTR1 TEs and generates a .fa and submits for clustering
                calls name_adder_for_mafft.R internally 
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains, then ./$name and ../ cyclically
    - Example usage: bash path-to-scripts/better-bed_to_tree.sh 
    - Outputs: whole-IAPLTR1-$name.bed (with ID-names), whole-IAPLTR1-$name.fa, and outputs of run-mafft-cluster.sh

better-IAPLTR-1d1-puller.sh

    - Synopsis: Open script for more extensive documentation. Takes in a fasta.out and returns solo, whole-1d1, and whole-non1d1 bed6's looping through mouse-strains-list.txt. Produces counts for each in IAPLTR-counts.txt 
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains, then ./$name and ../ cyclically 
    - Example usage: bash path-to-scripts/better-IAPLTR-1d1-puller.sh
    - Outputs: all-IAPLTR+Int-$name.fasta.out, all-IAPLTR+Int-$name.bed, solo+whole-IAPLTR-$name.bed, solo-IAPLTR-$name.bed, whole-IAPLTR-$name.bed, whole-non1d1-IAPLTR-$name.bed, whole-1d1-IAPLTR-$name.bed, IAPLTR-counts.txt

! blast-cleaner.sh [DEPRECATED: use minimap_and_syri.sh and liftOver.sh]

    - Synopsis: takes blast output file and reformats, checks for nearby LTR1s and returns a 1 if yes, 0 if no. loops through mouse-strains-list.txt
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains
    - Example usage: bash path-to-scripts/blast-cleaner.sh 
    - Outputs: $ref_name-$name-blastcount.txt

! blast.sh [DEPRECATED: use minimap_and_syri.sh and liftOver.sh] 

    - Synopsis: runs blastn on mm39 to CAST to generate alignment profile
    - Directory info: uses cwd
    - Example usage: bash path-to-scripts/blast.sh 
    - Outputs: mm39-blasted-Cast-specific-peaks.txt

! C3H-blast-looper.sh [DEPRECATED: use minimap_and_syri.sh and liftOver.sh]

    - Synopsis: BLASTs 1000bp upstream or downstream of IAPLTR1s, depending on if its on the 3' or 5' end. Loops through all strain directories against refname, which is passed in
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains, then ./$name and ../ cyclically
    - Example usage: bash path-to-scripts/C3H-blast-looper.sh SPRET_EiJ
    - Outputs: $name/blast-$refname-$name.txt

! C3H_DBA_blast-tester.sh [DEPRECATED: use minimap_and_syri.sh and liftOver.sh]

    - Synopsis: runs blastn for C3H to DBA 
    - Directory info: uses cwd
    - Example usage: bash path-to-scripts/C3H_DBA_blast-tester.sh 
    - Outputs: blast-C3H_HeJ-DBA_2J.txt    

Chi-squared.R

    - Synopsis: Called in DEG-enrichment.sh. Takes in-matrix.txt and performs a Chi-squared significance test. returns an outfile of p-values according to the observed and control counts produced by DEG-enrishment.sh
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/Chi-squared.R 
    - Outputs: outfile.txt

DEG-enrichment.sh

    - Synopsis: Takes in a reference (strain1) and query (strain2) genome and requires $strain2.liftoff.ase_bed.txt, LTR1-$strain1-to-$strain2-mapping.bed, and $strain2.VM_IAP.bed 
                Extracts DEGs from ase_bed file, pulls shared and strain specific LTR1s near DEGs, pulls VM-IAPLTR1 TEs near DEGs, then runs a randomization significance test compared to shuffled query genome
                Calls Chi-squared.R and p_val-averager.R internally
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/T2T/chainfiles/mapping-cast-Bl6-transcripts
    - Example usage: bash path-to-scripts/DEG-enrichment.sh T2T_CAST_EiJ T2T_C57BL_6J 
    - Outputs: $strain2.DEGs.bed, $strain2.TSS_DEGs.bed, ss-LTR1-$strain1-to-$strain2-mapping.bed, shared-LTR1-$strain1-to-$strain2-mapping.bed, ss-TSS_DEG-$strain1-to-$strain2.bed, shared-TSS_DEG-$strain1-to-$strain2.bed
               $strain2.LTR1-VM-test.bed, $strain2.VM_IAPLTR1s.bed, $strain2.non-VM_IAPLTR1s.bed, VM_IAPLTR1s-TSS_DEG-$strain2.bed, non-VM_IAPLTR1s-TSS_DEG-$strain2.bed, 
               ss-counts.txt, shared-counts.txt, VM-counts.txt, non-VM-counts.txt, ss-control-counts.txt, shared-control-counts.txt, VM-control-counts.txt, non-VM-control-counts.txt,
               p_val-matrix.txt, avg-p_vals.txt

fasta_to_tree_loop.sh

    - Synopsis: Loops fasta_to_tree.sh through mouse-strains-list.txt 
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains
    - Example usage: bash path-to-scripts/fasta_to_tree_loop.sh
    - Outputs: see fasta_to_tree.sh

fasta_to_tree.sh 

    - Synopsis: Clusters + turn into a tree using run-mafft-cluster.sh and tree_grower.R, respectively. Submits to hpc via submit-job-cclake.sh
    - Directory info: uses cwd
    - Example usage: bash path-to-scripts/fasta_to_tree.sh strain-name
    - Outputs: see run-mafft-cluster.sh or tree_grower.R 

gff3_to_bed6plusASE.sh

    - Synopsis: Takes a gff3 file and converts it into a bed6 plus ase information as a 7th additional column
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/T2T/chainfiles/mapping-cast-Bl6-transcripts
    - Example usage: bash path-to-scripts/gff3_to_bed6plusASE.sh path-to-gff3file path-to-asefile
    - Outputs: $name.bed, $name.ase_bed.txt  

! IAPLTR-1d1-puller.R [DEPRECATED: use better-IAPLTR-1d1-puller.sh]

    - Synopsis: DOES NOT WORK. Intended to perform same task as better-IAPLTR-1d1-puller.sh via R
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/IAPLTR-1d1-puller.R 
    - Outputs: outfile.all.txt, outfile.solos.txt, outfile.fulls.txt, outfile.1d1.txt, outfile.non1d1.txt

! IAPLTR1-mafft.sh [DEPRECATED: use better-bed_to_tree.sh]

    - Synopsis: Loops through mouse-strains-list, generates a .fa from a bed6 and submits run-mafft-cluster.sh to hpc 
    - Directory info: uses cwd
    - Example usage: bash path-to-scripts/IAPLTR1-mafft.sh
    - Outputs: $name/all-IAP-$name.fa, see run-mafft-cluster.sh

IAPLTR-barchart.R 

    - Synopsis: Takes in IAPLTR-counts.txt as in-file.txt and returns a barchart of solo, full, and 1d1 counts for LTR1's, 1a's, and 2's
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/IAPLTR-barchart.R  
    - Outputs: outfile.png

! IAPLTR-counter.sh [DEPRECATED: use better-IAPLTR-1d1-puller.sh]

    - Synopsis: Intended for use with outputs of 1delta1_reader.R. Produces outfile similar to IAPLTR-counts.txt
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains
    - Example usage: bash path-to-scripts/IAPLTR-counter.sh
    - Outputs: IAPLTR-counts.txt

liftOver.sh

    - Synopsis: Generates file of 1kb flanks near LTRs, generates chainfile to then generate LTR1-$strain1-to-$strain2-mapping.bed. Requires vcf file from minimap_and_syri.sh
                Calls LTR_regions_puller.R internally
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains/liftOver/T2T
    - Example usage: bash path-to-scripts/liftOver.sh reference-genome-name query-genome-name
    - Outputs: LTR1-flanks-$strain1.bed, LTR1-flanks-$strain2.bed, $strain1-to-$strain2.chain, $strain2-to-$strain1.chain, liftOver-LTR1-$strain1-to-$strain2.bed LTR1-$strain1-to-$strain2-unmapped_output,
               liftOver-LTR1-$strain2-to-$strain1.bed, LTR1-$strain2-to-$strain1-unmapped_output, LTR1-$strain1-to-$strain2-mapping.bed, LTR1-$strain2-to-$strain1-mapping.bed

! LTR1-region-puller.sh [DEPRECATED: use liftOver.sh]

    - Synopsis: Loops through mouse-strains-list.txt and generates bed6 of 1kb flanks near LTR1s and generates .fa
    - Directory info: sets cwd to cd ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains/$name 
    - Example usage: bash path-to-scripts/LTR1-region-puller.sh 
    - Outputs: IAPLTR1-flanks-$name.fa, all-EzInt-$name.bed, IAPLTR1-flanks-$name.bed

LTR_regions_puller.R

    - Synopsis: Reads in a full TE bed6 and returns 1000kb up or downstream of the LTR
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/LTR_regions_puller.R
    - Outputs: outfile.txt

mafft_heatmap_converter.sh

    - Synopsis: Loops through mouse-strains-list.txt and runs reformatting + NT -> number conversion for heatmap making, to be used in tree-grower.R
                Calls trimal_to_heatmap.R internally
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains
    - Example usage: bash path-to-scripts/mafft_heatmap_converter.sh 
    - Outputs: converted-trimal-$name.txt
    
! mafft_heatmap_readier.R [DEPRECATED: use trimal_to_heatmap.R]

    - Synopsis: converts trimal output into a format usable by tree-grower.R to generate MSA plot. Functionally similar to trimal_to_heatmap.R, but slower 
    - Directory info: uses cwd
    - Example usage: conda activate -n R path-to-scripts/mafft_heatmap_readier.R
    - Outputs: outfile.txt
    
make-your-bed.sh

    - Synopsis: Simple example of how to convert a fasta.out into a bed6. Would not recommend for practical use
    - Directory info: uses cwd
    - Example usage: bash path-to-scripts/make-your-bed.sh 
    - Outputs: $name.bed
    
minimap_and_syri.sh

    - Synopsis: Runs alignment for two genomes via minimap2 and then syri for VCF files (to later become chain files). Submitted to hpc by submit-minimap_syri-cclake.sh
                NOTE: Currently nonfunctional as of 11/8/25 due to Python dependancy issues in syri_env conda enviornment
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains/
    - Example usage: bash path-to-scripts/minimap_and_syri.sh reference-genome-name query-genome-name 
    - Outputs: $refgenome-$qrygenome-out.sam and a vcf file [don't know the name since syri sucks ass >:\ ]
        
multi_strain-tree_grower.R

    - Synopsis: tree-grower.R varient for multiple strain usage. Takes in phyml tree output (in-tree.txt), trimal_to_heatmap.R output (in-fasta.txt), and 1d1-aligner-for-all-subspecies-tree.sh output of 1d1's (in-Id1_matrix.txt)
                Returns a composite plot of a phylogenetic tree, MSA, and 1d1 concentration plot all on one figure
                NOTE: also has been used to plot populations individually using custom ylims noted on the bottom of the R file
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/multi_strain-tree_grower.R
    - Outputs: LTR_phylogeny-all_subspecies.png
        
name_adder_for_mafft.R

    - Synopsis: Takes a bed6 and converts columns 4 and 5 into custom names (either by strain or by LTR type) for use with mafft during clustering
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/name_adder_for_mafft.R
    - Outputs: outfile.txt
        
newick-rel_length-puller.sh

    - Synopsis: Takes in all_subspecies.phylip_phyml_tree_all_subspecies.txt and returns the relative lengths of each branch in the tree
                NOTE: originally planned for use to determine % sequence homology, but is relatively useless in that goal
    - Directory info: uses cwd
    - Example usage: bash path-to-scripts/newick-rel_length-puller.sh 
    - Outputs: newick_distcounts_normed.txt

only_syri.sh

    - Synopsis: Variant of minimap_and_syri.sh that only runs syri to generate the .vcf file from the .sam file. Mode for troubleshooting reasons
                NOTE: Currently nonfunctional as of 11/8/25 due to Python dependancy issues in syri_env conda enviornment
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains/ 
    - Example usage: bash path-to-scripts/only_syri.sh reference-genome-name query-genome-name 
    - Outputs: a vcf file 

pie-maker.R

    - Synopsis: Takes in all_subspecies-population-list.txt from ../hybrids and returns a pie chart containing each strain's prevalence in each population (in reference to all_subspecies tree plot)
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/hybrids to function properly
    - Example usage: conda activate -n R Rscript path-to-scripts/pie-maker.R 
    - Outputs: Pop-pie.png (I am very funny)
            
progress_fiddling.sh

    - Synopsis: Varient of gff3_to_bed6plusASE.sh that contains a custom-made progress bar for aligning bed6 and ase info
                NOTE: intended as practice, but progress bar can be copied into other scripts 
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/T2T/chainfiles/mapping-cast-Bl6-transcripts
    - Example usage: see ff3_to_bed6plusASE.sh 
    - Outputs: see ff3_to_bed6plusASE.sh
            
p_val-averager.R

    - Synopsis: Takes in p_val-matrix.txt from Chi-squared.R and returns a table of the average of each column. Integrated into DEG-enrichment.sh
    - Directory info: uses cwd (which is set by DEG-enrichment.sh)
    - Example usage: conda activate -n R Rscript path-to-scripts/p_val-averager.R 
    - Outputs: p_val-averages.txt
            
randomization-test-p_val.R

    - Synopsis: unused variant of Chi-squared.R which converts a table of chi-squared test statistics named in-file.txt into p-values
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/randomization-test-p_val.R 
    - Outputs: outfile.txt
           
! run-1delta1.sh [DEPRECATED: use better-IAPLTR-1d1-puller.sh ]

    - Synopsis: intended as a looper for 1delta1_reader.R through mouse-strains-list.txt
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/ to function properly 
    - Example usage: bash path-to-scripts/run-1delta1.sh 
    - Outputs: see 1delta1_reader.R  
               
run-mafft-cluster.sh

    - Synopsis: takes in a .fa and returns a phylogenetic tree based on sequence homology as well as an ordered list of each tip label. Submitted to hpc by submit-job-cclake.sh
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/ to function properly  
    - Example usage: conda activate -n mafft_conda scripts/run-mafft-cluster.sh strain-name
    - Outputs: mafft-$name.fa, trimal-$name.fa, $name.phylip, temp-$name.out, $name.phylip_phyml_stats_$name.txt, $name.phylip_phyml_tree_$name.txt, $name-phylip_phyml_tree-list.txt, $name-phylip_phyml_tree-list.txt
               
NOTE: the variants of run-mafft-cluster.sh and thier respective submission scripts exist due to troubleshooting reasons. Functionally, they are identical to run-mafft-cluster.sh and submit-job-cclake.sh

run-mafft-ERVB7_LTRs.sh

    - Synopsis: Variant of run-mafft-cluster.sh for clustering of ERVB7_LTRs. Submitted to hpc by submit-ERVB7_LTRs-cclake.sh
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/hybrids to function properly  
    - Example usage: conda activate -n mafft_conda ../scripts/run-mafft-ERVB7_LTRs.sh ERVB7_LTRs 
    - Outputs: see run-mafft-cluster.sh
               
run-mafft-SPRETs.sh

    - Synopsis: Variant of run-mafft-cluster.sh for clustering of SPRET LTR1s, 1a's, and 2s. Submitted to hpc by submit-SPRETmafft-cclake.sh
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/SPRET_EiJ to function properly  
    - Example usage: conda activate -n mafft_conda ../scripts/run-mafft-SPRETs.sh what-to-cluster
    - Outputs: see run-mafft-cluster.sh 
                   
run-mafft-whole_IAPLTR1.sh

    - Synopsis: Variant of run-mafft-cluster.sh for clustering of whole IAPLTR1s (LTRs + TEs). Submitted to hpc by submit-wIAPLTR1-cclake.sh
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/hybrids to function properly  
    - Example usage: conda activate -n mafft_conda ../scripts/run-mafft-whole_IAPLTR1.sh what-to-cluster 
    - Outputs: see run-mafft-cluster.sh
                   
submit-ERVB7_LTRs-cclake.sh 

    - Synopsis: Submits run-mafft-ERVB7_LTRs.sh to hpc
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/hybrids to function properly 
    - Example usage: sbatch ../scripts/submit-ERVB7_LTRs-cclake.sh
    - Outputs: see run-mafft-cluster.sh
                   
submit-job-cclake.sh

    - Synopsis: Submits run-mafft-cluster.sh to hpc
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains to function properly 
    - Example usage: sbatch ../scripts/submit-ERVB7_LTRs-cclake.sh what-to-cluster
    - Outputs: see run-mafft-cluster.sh

submit-minimap_syri-cclake.sh

    - Synopsis: Submits minimap_and_syri.sh to hpc
    - Directory info: minimap_and_syri.sh will set cwd to ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains 
    - Example usage: sbatch path-to-scripts/submit-minimap_syri-cclake.sh reference-genome-name query-genome-name
    - Outputs: see minimap_and_syri.sh

submit-SPRETmafft-cclake.sh

    - Synopsis: Submits run-mafft-SPRETs.sh to hpc
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/SPRET_EiJ to function properly 
    - Example usage: sbatch ../scripts/submit-SPRETmafft-cclake.sh what-to-cluster
    - Outputs: see run-mafft-cluster.sh   

submit-syri-cclake.sh

    - Synopsis: Submits only_syri.sh to hpc
    - Directory info: only_syri.sh will set cwd to ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains 
    - Example usage: sbatch path-to-scripts/submit-only_syri-cclake.sh reference-genome-name query-genome-name
    - Outputs: see only_syri.sh

submit-wIAPLTR1-cclake.sh

    - Synopsis: Submits run-mafft-whole_IAPLTR1.sh to hpc 
    - Directory info: must be in ~/rds/rds-afs-lab-rds-FxufPjywHiE/genomes/Mouse-strains/hybrids to function properly  
    - Example usage: sbatch ../scripts/submit-submit-wIAPLTR1-cclake.sh 
    - Outputs: see run-mafft-cluster.sh      

tree_grower.R

    - Synopsis: generic version of multi_strain-tree_grower.R that only plots the phylogeny. Input is only the phyml tree output (in-tree.txt)
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/tree_grower.R 
    - Outputs: outfile.png
    
tree_grower.sh

    - Synopsis: Loops through mouse-strains-list.txt and runs tree_grower.R for each strain
    - Directory info: sets cwd to ~/rds/rds-acf1004-afs-lab-rds/genomes/Mouse-strains
    - Example usage: bash path-to-scripts/tree_grower.sh 
    - Outputs: $name/LTR_phylogeny_$name.png
    
trimal_to_heatmap.R

    - Synopsis: Better version of mafft_heatmap_readier.R. Converts a trimal output into a matrix with NTs seperated into rows with the tip label as the first column
    - Directory info: uses cwd
    - Example usage: conda activate -n R Rscript path-to-scripts/trimal_to_heatmap.R strain-name
    - Outputs: msa-strain-name.txt
